### PR TITLE
Add additional documentation about fields

### PR DESCRIPTION
**Pull request by sethrj**
This adds a few more classes to the user facing documentation and uploads any PR tagged as "documentation" rather than just PR titles that have `doc-` in them.

### CODE REVIEW

File: doc/main/api/celeritas.rst [hunk 1]
```diff
@@ -75,6 +75,51 @@ On-device access
...
+  state and the templated field.
+Integrator
+  Numerically integrates a new position/momentum state given the start,
+  path derivative, and step length.
+Driver
+  Integrate path segments that satisfy certain error conditions, solving for
+  the required segment length.
+Propagator
+  Given a maximum physics step, advance the geometry state and momentum along
+  the field lines, satisfying given constraints for the maximum geometry error.
```
Comment (pcanal) [inline doc/main/api/celeritas.rst]: Do we document somewhere what is `certain error conditions` and how to give the 'constraints'?

Comment (sethrj) [reply doc/main/api/celeritas.rst]: The detailed field propagator documentation gives more details. I can also link to the field propagator options.

Comment (pcanal) [reply doc/main/api/celeritas.rst]: > I can also link to the field propagator options.

+1

File: .github/workflows/doc.yml [hunk 2]
```diff
@@ -7,6 +7,14 @@ concurrency:
   group: doc-${{github.ref}}-${{github.event.pull_request.number || github.run_number}}-${{github.workflow}}
   cancel-in-progress: true
 
+env:
+  celer_upload_doc:  >-
+    ${{  github.ref_name == 'develop'
+      || contains(github.ref_name, 'backports/')
+      || contains(github.head_ref, 'release-')
+      || contains(github.event.pull_request.labels.*.name, 'documentation')
```
Comment (pcanal) [inline .github/workflows/doc.yml:15]:
```suggestion
    ${{  github.ref_name == 'develop'             # push into develop 
      || contains(github.ref_name, 'backports/')  # push into backport
      || contains(github.head_ref, 'release-')    # pull request preparing release
      || contains(github.event.pull_request.labels.*.name, 'documentation')
```

**Review by pcanal — APPROVED** [review]
LGTM.

File: .github/workflows/doc.yml (hunk 2 above)
Comment (sethrj) [reply .github/workflows/doc.yml:15]: Alas comments are included as literals so it doesn't work inside this.

**Pull request merged**